tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+rng = range(np.where(np.floor(freqs) == 1.)[0][0],
+            np.where(np.ceil(freqs) == fmax - 1)[0][0])
+
+psds_plot = 10 * np.log10(psds)
+psds_mean = psds_plot.mean((0, 1))[rng]
+psds_std = psds_plot.std((0, 1))[rng]
+axes.plot(freqs[rng], psds_mean, color='b')
+axes.fill_between(freqs[rng], psds_mean - psds_std, psds_mean + psds_std,
+                  color='b', alpha=.5)
+axes.set(title="PSD spectrum", xlabel='Frequency [Hz]',
+         ylabel='Power Spectral Density [dB]')
+plt.xlim([0, fmax])
+fig.show()
+
+##############################################################################
+# SNR spectrum - trial average
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+# Average over trials
+axes.plot(freqs, snrs.mean(axis=0).T, color='b')
+# Average over trials and channels
+axes.plot(freqs, snrs.mean(axis=0).mean(axis=0), color='r')
+axes.set(title="SNR spectrum - trial average", xlabel='Frequency [Hz]',
+         ylabel='SNR', ylim=[0, 20])
+fig.show()
+
+##############################################################################
+# SNR spectrum - channel average
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+fig, axes = plt.subplots(1, 1, sharex='all', sharey='all', dpi=300)
+# Average over channels
+axes.plot(freqs, snr_spectrum(psds[:, :, :], 3, 1).mean(axis=1).T, color='b')
+# Average over trials and channels
+axes.plot(freqs, snr_spectrum(psds[:, :, :], 3, 1).mean(axis=0).mean(axis=0),
+          color='r')
+axes.set(title="SNR spectrum - channel average", xlabel='Frequency [Hz]',
+         ylabel='SNR', ylim=[0, 20])
+fig.show()
+
+
+###############################################################################
+# Subsetting data
+# ---------------
+#
+# For statistical comparison you probably want specific subsets of the SNR
+# array. Either some channels, or - obviously - different trials depending on
+# the stimuli.
+#
+# - So far, one needs to define the indices of the channels / trials by hand -
+#   not nice.
+# - Alternatively, one can subset trials already at the epoch level using MNEs
+#   event information, and create individual PSD and SNR objects.
+#
+# Here we have already subsetted trials before snr calculation (only 12Hz
+# stimulation) and will now compare SNR in different channel subsets.
+#
+# For illustration purposes, we will still subset the first 5 and last 5 of the
+# 10 trials with 12hz stimulation.
+
+
+##############################################################################
+# Define ROIs
+roi_temporal = ['T7', 'F7', 'T8', 'F8']  # temporal
+roi_aud = ['AFz', 'Fz', 'FCz', 'Cz', 'CPz', 'F1', 'FC1',
+           'C1', 'CP1', 'F2', 'FC2', 'C2', 'CP2']  # auditory roi
+roi_vis = ['POz', 'Oz', 'O1', 'O2', 'PO3', 'PO4', 'PO7',
+           'PO8', 'PO9', 'PO10', 'O9', 'O10']  # visual roi
+
```

Comment:
i recently wrote code to visualize SNR for all channels on a topoplot of the head... 

looks like this, for grand average assr in a different dataset: 
<img width="347" alt="grafik" src="https://user-images.githubusercontent.com/33089761/95067566-53ac5a80-0704-11eb-9931-2454d7f3b5b4.png">

we could include this here as well.. 
doesnt carry that much info, but might be nice company for the roi part

what do you think?

Reply:
Looks good. Let's add this. 

Again, I would do it later though, when we are happy with the substantial stuff.